slack_sdk/socket_mode/aiohttp/__init__.py

```diff
@@ -126,70 +136,124 @@ def __init__(
+                            )
+                            should_reconnect = True
+
+                        if self.last_ping_pong_time is not None:
+                            disconnected_seconds = int(
+                                time.time() - self.last_ping_pong_time
+                            )
+                            if disconnected_seconds >= (self.ping_interval * 4):
+                                self.logger.info(
+                                    "The connection seems to be stale. Reconnecting..."
+                                    f" reason: disconnected for {disconnected_seconds}+ seconds)"
+                                )
+                                self.stale = True
+                                self.last_ping_pong_time = None
+                                should_reconnect = True
+
+                        if should_reconnect is True or not await self.is_connected():
+                            await self.connect_to_new_endpoint()
+
+                except Exception as e:
+                    self.logger.error(
+                        "Failed to check the current session or reconnect to the server "
+                        f"(error: {type(e).__name__}, message: {e})"
                     )
-                    await self.connect_to_new_endpoint()
-            except Exception as e:
-                self.logger.error(
-                    "Failed to check the current session or reconnect to the server "
-                    f"(error: {type(e).__name__}, message: {e})"
+        except asyncio.CancelledError:
+            if self.trace_enabled:
+                self.logger.debug(
+                    "The running monitor_current_session task is now cancelled"
                 )
+            raise
 
     async def receive_messages(self) -> None:
-        consecutive_error_count = 0
-        while not self.closed:
-            try:
-                message: WSMessage = await self.current_session.receive()
-                if self.logger.level <= logging.DEBUG:
-                    type = WSMsgType(message.type)
-                    message_type = type.name if type is not None else message.type
-                    message_data = message.data
-                    if isinstance(message_data, bytes):
-                        message_data = message_data.decode("utf-8")
-                    self.logger.debug(
-                        f"Received message (type: {message_type}, data: {message_data}, extra: {message.extra})"
-                    )
-                if message is not None:
-                    if message.type == WSMsgType.TEXT:
+        try:
+            consecutive_error_count = 0
+            while not self.closed:
+                try:
+                    message: WSMessage = await self.current_session.receive()
+                    if self.trace_enabled and self.logger.level <= logging.DEBUG:
+                        type = WSMsgType(message.type)
```

Comment:
The next conditional below checks that `message` is not `None` - should that be done here too?

Reply:
The type should always exists but just in case, we may want to have additional check before the assignment. will update this part later.